### PR TITLE
add org.kde.okular

### PR DESCRIPTION
**Pull request by nedrichards**
Adds Okular document reader

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.kde.okular

**Comment by nedrichards** [comment]
@aleixpol as with the other KDE apps, your review is always appreciated.

**Comment by nedrichards** [comment]
@tsdgeos the same, and obviously as with all flathub apps, upstream gets commit access and can release it any way they wish.

File: org.kde.okular.json [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+            "sources": [ 
+                { 
+                    "type": "archive", 
+                    "url": "https://libzip.org/download/libzip-1.4.0.tar.xz", 
+                    "sha256": "e508aba025f5f94b267d5120fc33761bcd98440ebe49dbfe2ed3df3afeacc7b1" 
+                } 
+            ]
+        },
+        {
+            "name": "chmlib",
```
Comment (nedrichards) [inline org.kde.okular.json]: this needs a reconfigure for aarch64 https://flathub.org/builds/#/builders/41/builds/95

**Comment by nedrichards** [comment]
bot, build org.kde.okular

**Comment by nedrichards** [comment]
bot, build org.kde.okular

File: org.kde.okular.json [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://sourceforge.net/projects/boost/files/boost/1.63.0/boost_1_63_0.tar.bz2",
+                    "sha256": "beae2529f759f6b3bf3f4969a19c2e9d6f0c503edcb2de4a61d1428519fcb3b0"
+                }
+            ]
+        },
+        {
+            "name": "kjs",
+            "config-opts": ["-DCMAKE_BUILD_TYPE=Release", "-DCMAKE_INSTALL_LIBDIR=lib"],
```
Comment (TingPing) [inline org.kde.okular.json]: `-DCMAKE_INSTALL_LIBDIR=lib` isn't needed any more?

Comment (nedrichards) [reply org.kde.okular.json]: You're right, I should do a pr to the flatpak-kde-applications repo as well.

File: org.kde.okular.json [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        {
+            "name": "openjpeg",
+            "buildsystem": "cmake-ninja",
+            "config-opts": [
+                "-DCMAKE_BUILD_TYPE=Release"
+            ],
+            "cleanup": [
+                "/bin",
+                "/lib/openjpeg-*",
+                "*.a"
```
Comment (barthalion) [inline org.kde.okular.json]: I'd move it to top-level cleanup.

**Comment by barthalion** [comment]
These recent additions of KDE apps is at least partially upstream initiative?

**Comment by nedrichards** [comment]
@barthalion some are pure upstream, some 'packagers' others start as packaging but have upstream comitters as well. It's a mix.

For my own part I want to be able to migrate the KDE apps that Endless offers to users to being provided by Flathub, so I'm interested in helping out where I can.

File: org.kde.okular.json (hunk 3 above)
Comment (nedrichards) [reply org.kde.okular.json]: Makes sense.

File: org.kde.okular.json (hunk 2 above)
Comment (eszlari) [reply org.kde.okular.json]: For projects that use https://github.com/KDE/extra-cmake-modules` (like all of KDE) `-DCMAKE_INSTALL_LIBDIR=lib` is still needed.

Comment (aleixpol) [reply org.kde.okular.json]: @eszlari do you know what's the issue? I could look into it if you can tell me what's going wrong.

Comment (eszlari) [reply org.kde.okular.json]: I don't know what the cause is, but without it libs get installed under `lib64`. See my last comment under https://github.com/flatpak/freedesktop-sdk-base/issues/15

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.kde.okular

**Pull request closed without merging**